doc/tutorials/sgx_ virtualization.rst

```diff
@@ -0,0 +1,213 @@
+
+Instructions 
+============
+
+By default, SGX support for Guest is not enabled on the Master branch. Thus,
+you must perform the following:
+
+#. Make sure SGX is supported on the platform (CPUID.07H.EBX[2] should be 1).
+#. Opt-in to the SGX feature in the BIOS. Follow these instructions: 
+
+   a) Go to the Security page:
+
+      .. figure:: images/sgx-2.jpg
+         :width: 500px
+         :align: left
+
+   #) Enable SGX and configure the SGX Reserved Memory size as below: 
+
+      * Intel Software Guard Extension (SGX) -> Enabled 
+      * SGX Reserved Memory Size -> 128MB
+
+      .. figure:: images/sgx-3.jpg
+         :width: 500px
+         :align: left
+
+      .. note:: 
+         Not all SGX Reserved Memory can be used as EPC. On KBL-NUC-i7,
+         the SGX EPC size is 0x5d80000 (93.5MB) when the SGX Reserved Memory Size is set to 128MB. 
+
+#. Add the EPC config in the VM configuration:
+   Apply the patch to enable SGX support in UOS in the SDC scenario:
+   https://github.com/binbinwu1/acrn-hypervisor/commit/0153b2b9b9920b61780163f19c6f5318562215ef.patch
+
+#. Enable SGX in Guest:
+
+   * Refer to https://github.com/intel/linux-sgx on how to enable SGX in Linux Guest.
+   * Refer to https://software.intel.com/en-us/articles/getting-started-with-sgx-sdk-for-windows on how to enable SGX in Windows Guest.
+
+SGX Capability Exposure
+***********************
+
+CPUID Virtualization
+====================
+
+CPUID Leaf 07H
+--------------
+
+* CPUID_07H.EAX[2] SGX: Supports Intel Software Guard Extensions if 1. If SGX is supported in Guest, this bit will be set.
+* CPUID_07H.ECX[30] SGX_LC: Supports SGX Launch Configuration if 1. Currently, ACRN does not support the SGX Launch Configuration. This bit will not be set. Thus, the Launch Enclave must be signed by the Intel SGX Launch Enclave Key.
+
+CPUID Leaf 12H
+--------------
+
+**Intel SGX Capability Enumeration**
+
+* CPUID_12H.0.EAX[0] SGX1: If 1, indicates that Intel SGX supports the collection of SGX1 leaf functions.If is_sgx_supported and the section count is initialized for the VM, this bit will be set.
+* CPUID_12H.0.EAX[1] SGX2: If 1, indicates that Intel SGX supports the collection of SGX2 leaf functions. If hardware supports it and SGX enabled for the VM, this bit will be set.
+* Other fields of CPUID_12H.0.EAX aligns with the physical CPUID.
+
+**Intel SGX Attributes Enumeration**
+
+* CPUID_12H.1.EAX & CPUID_12H.1.EBX aligns with the physical CPUID.
+* CPUID_12H.1.ECX & CPUID_12H.1.EDX reflects the allow-1 setting in the Extended feature (same structure as XCR0).
+
+The hypervisor may change the allow-1 setting of XFRM in ATTRIBUTES for VM. 
+If some feature is disabled for the VM, the bit is also cleared, eg. MPX.
+
+**Intel SGX EPC Enumeration**
+
+* CPUID_12H.2: The hypervisor presents only one EPC section to Guest. This vcpuid value will be constructed according to the EPC resource allocated to Guest.
+
+MSR Virtualization
+==================
+
```

Comment:
Needs an introduction here to explain what these sections are describing.  I *think* these are implementation notes about what the hypervisor does.